### ISSUE TITLE
TypeScript errors in `editor.api.d.ts` typings

### ISSUE THREAD

**Issue by timkendrick** [Microsoft/monaco-editor#757]
Really excited to be using the new webpack build of monaco-editor! I've been trying to use it as part of a library that emits type declarations, and it looks like the shipped typings are broken – see below for details (and a possible fix?)

monaco-editor version: 0.11.0
Browser: N/A
OS: macOS 10.13.3
Steps or JS usage snippet reproducing the issue:
```shell
mkdir temp && cd temp
npm init -y
npm install -D typescript monaco-editor

# Write a simple index file that reexports some of the monaco types
echo "export { Selection } from 'monaco-editor';" > index.ts

# Attempt to compile with declarations
./node_modules/.bin/tsc --declaration index.ts
```

Compilation fails with following errors:

```
node_modules/monaco-editor/esm/vs/editor/editor.api.d.ts(766,1): error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
node_modules/monaco-editor/esm/vs/editor/editor.api.d.ts(5295,43): error TS2503: Cannot find namespace 'monaco'.
node_modules/monaco-editor/esm/vs/editor/editor.api.d.ts(5296,43): error TS2503: Cannot find namespace 'monaco'.
```

...it seems this is related to [this commit](https://github.com/Microsoft/monaco-editor/commit/c6192de715bf1712e64c7f1c849ee52e2270a154) that introduces some manual text manipulation to the `monaco.d.ts` file and saves it as `editor.api.d.ts`.

I'm not sure exactly what's going on here, but I got the above example to compile properly by ignoring the `editor.api.d.ts` and doing the following:

```shell
# Add "export = monaco;" to the end of monaco.d.ts
echo "export = monaco;" >> node_modules/monaco-editor/monaco.d.ts

# Change the package.json to use the typings in monaco.d.ts, not editor.api.d.ts
sed -i .bak 's/esm\/vs\/editor\/editor.api.d.ts/monaco.d.ts/' node_modules/monaco-editor/package.json

# Now compiles fine!
./node_modules/.bin/tsc --declaration index.ts
```

**Comment by corbinu** [comment]
I am seeing this exact error

**Comment by alexandrudima** [comment]
Thanks for the heads up. Just published 0.11.1

**Comment by timkendrick** [comment]
Works great, thanks for the speedy fix!
